libobs/obs-encoder.c

```diff
@@ -221,7 +241,7 @@ static void remove_connection(struct obs_encoder *encoder, bool shutdown)
 		audio_output_disconnect(encoder->media, encoder->mixer_idx,
 					receive_audio, encoder);
 	} else {
-		if (gpu_encode_available(encoder)) {
+		if (os_atomic_load_long(&obs->video.gpu_encoder_active)) {
```

Comment:
The commit that change this may need more explanation about it.

Reply:
To be honest I don't even remember why this change was needed, but it's definitely more correct than just relying on gpu encoding being active if it is available.

Reply:
Oh I remember why now: There might be encoders for which `gpu_encode_available` returns false due to the platform not supporting them, but that also don't have a `.encode` function as a fallback.
In that case starting the encoding fails, and it's better to fail gracefully in that case rather than to potentially do something bad.

Reply:
I spoke to @openglfreak and it looks like he isn't likely to finish this PR, but would be happy for me to rebase it. 

If I rebase, all the comment history will be lost, as far as I understand, so first I wanted to check if this can easily be resolved as is - hopefully saving work for everyone involved. @tytan652, what are your thoughts?

Reply:
Feel free to open a new PR and link to this one. We can close this PR (so it's off our pile of stuff to review), but the comment history would still be preserved.